src/key.rs

```diff
@@ -198,36 +199,17 @@ impl PublicKeyParts for RsaPublicKey {
 }
 
 impl PublicKey for RsaPublicKey {
-    fn encrypt<R: CryptoRngCore>(
+    fn encrypt(
         &self,
-        rng: &mut R,
-        padding: PaddingScheme,
+        rng: &mut impl CryptoRngCore,
+        padding: impl PaddingScheme,
```

Comment:
Why did you go for `impl` over a generic? I have been bitten by libraries which used `impl` because you can't provide type hints directly, so I am sceptical on this

Reply:
I can add the explicit generic parameters back, however it is possible to type hint with the arguments rather than the generic parameters:

https://github.com/RustCrypto/RSA/pull/244/files#diff-0988d5d2259e1a401001675578c38d09c46290b877b66b07fb96a7e034ccdf70R434

Reply:
but that only works if working with `Option` right?

Reply:
You can always use `as` as a sort of pseudo type ascription as well, e.g.:

```rust
None as Option<&mut DummyRng>
```

works in addition to

```rust
Option::<&mut DummyRng>::None
```

...but I can go ahead and change the `impl Trait` usages back to explicit parameters

Reply:
befaa734e06c78a9f8c686616effc0cd36b902af switches things back to generic parameters

Reply:
@tarcieri I am confused, the latest shows `impl` again

Reply:
Oh sorry, https://github.com/RustCrypto/RSA/commit/befaa734e06c78a9f8c686616effc0cd36b902af fixed some of the cases but not all of them. Will do another pass.